tiles/src/main/java/com/protomaps/basemap/feature/CountryCoder.java

```diff
@@ -0,0 +1,84 @@
+package com.protomaps.basemap.feature;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.onthegomap.planetiler.geo.GeoUtils;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import org.locationtech.jts.geom.*;
+import org.locationtech.jts.index.strtree.STRtree;
+
+public class CountryCoder {
```

Comment:
You might want to use Planetiler's `PolygonIndex`, see https://github.com/onthegomap/planetiler/blob/f5fe38a4d4400f3aa30998a1eb3ac551ce3e47d9/planetiler-core/src/main/java/com/onthegomap/planetiler/geo/PolygonIndex.java#L20, cc @msbarry

Reply:
We may need to enhance that to handle (Polygon contains LineString) - looks like Contains only supports Point and Intersects works for all geometries? TBD if it actually matters

Reply:
I was thinking about that, it seems like country coder should return the country that feature is "most within" so we could sample a few points? Here's the api that prepared geometry exposes when testing geometries: https://locationtech.github.io/jts/javadoc/org/locationtech/jts/geom/prep/PreparedGeometry.html